File: Phidget Code/src/Events/Event.java
package Events;

/**
 * Created by bri_e on 13-03-17.
 */
public class Event {

    private EventType eventType;
    private double time;

    public Event(){}

    public Event(double evenTime, EventType evnt){
        time = evenTime;
        eventType = evnt;
    }

    public Event(EventType event){
        eventType = event;
    }

    public double getTime(){
        return time;
    }

    public EventType getType(){
        return eventType;
    }

}

File: PhidgetScalaCode/src/Events/Event.scala
package Events

/**
  * Created by bri_e on 20-04-17.
  * Class containing the different case classes with different events
  */
abstract class Event(var eventTime:Long)

case class BarEvent(var evntTime:Long)           extends Event(evntTime)
case class DemoPhaseEvent(var evntTime:Long)     extends Event(evntTime)
case class HeatEvent(var evntTime:Long)          extends Event(evntTime)
case class LightEvent(var evntTime:Long)         extends Event(evntTime)
case class NewMatchPlanEvent(var evntTime:Long)  extends Event(evntTime)
case class PassageEvent(var evntTime:Long)       extends Event(evntTime)
case class StandEvent(var evntTime:Long)         extends Event(evntTime)
case class TurnEvent(var evntTime:Long)          extends Event(evntTime)
case class VibrationEvent(var evntTime:Long)     extends Event(evntTime)
